ttnn/cpp/ttnn/operations/experimental/paged_cache/device/paged_cache_operation.hpp

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <optional>
+
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/run_operation.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/compute_kernel_config.hpp"
+
+namespace ttnn::operations::experimental::paged_cache {
+
+enum class PagedUpdateCacheOpParallelizationStrategy {
+    MULTI_CORE
+};
+
+enum class PagedUpdateCacheOpType {
+    UPDATE, FILL
+};
+
+struct PagedUpdateCacheDeviceOperation {
+    const uint32_t batch_idx;
+    const std::vector<uint32_t> update_idxs;
+    const uint32_t batch_offset;
+    const PagedUpdateCacheOpType op_type;
+    const DeviceComputeKernelConfig compute_kernel_config;
+
+    PagedUpdateCacheOpParallelizationStrategy get_parallelization_strategy(const std::vector<Tensor> &input_tensors) const;
+
+    void validate(const std::vector<Tensor>& input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors) const;
+    const std::vector<tt::tt_metal::Shape> compute_output_shapes(
+        const std::vector<Tensor> &input_tensors) const;
+    std::vector<Tensor> create_output_tensors(
+        const std::vector<Tensor> &input_tensors) const;
+
+
+    operation::ProgramWithCallbacks create_program(
+        const std::vector<Tensor>& input_tensors,
+        const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+        std::vector<Tensor>& output_tensors
+    ) const;
+
+    static constexpr auto attribute_names =
+        std::forward_as_tuple("batch_idx", "update_idxs", "batch_offset", "op_type", "compute_kernel_config");
+
```

Comment:
afaik this is not needed anymore
Check with @arakhmati

Reply:
Akhmed: "after updating to C++20, we finally could start using https://github.com/qlibs/reflect. It lets us to iterate over the member variables of structs. So, we were able to delete a lot of the uses of attribute_names.
However, the library only works for simple structs without custom constructors, so we couldn't get rid of all the places where we use attribute_names yet. And therefore you might need it in some cases"